tests/generator/StringExtensionTests.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using NUnit.Framework;
+
+namespace GeneratorTests {
+	public class StringExtensionTests {
+		
+		public class KeywordsDataSource : IEnumerable
+		{
+			static string [] keywords = {
+				"abstract","event","new","struct","as","explicit","null","switch","base","extern",
+				"this","false","operator","throw","break","finally","out","true",
+				"fixed","override","try","case","params","typeof","catch","for",
+				"private","foreach","protected","checked","goto","public",
+				"unchecked","class","if","readonly","unsafe","const","implicit","ref",
+				"continue","in","return","using","virtual","default",
+				"interface","sealed","volatile","delegate","internal","do","is",
+				"sizeof","while","lock","stackalloc","else","static","enum",
+				"namespace",
+				"object","bool","byte","float","uint","char","ulong","ushort",
+				"decimal","int","sbyte","short","double","long","string","void",
+				"partial", "yield", "where"
+			};
+			
+			public IEnumerator GetEnumerator()
+			{
+				foreach (var key in keywords) {
+					yield return key;
+				}
+			}
+		}
+		
+		[TestCaseSource(typeof(KeywordsDataSource))]
+		public void KeywordsTest(string keyword)
+			=> Assert.AreNotEqual (keyword, keyword.GetSafeParamName (), "keyword");
+
+		[TestCase ("😀OhOh")]
+		[TestCase (" OhOh")]
```

Comment:
@haritha-mohan  has a point, "  OH_OH" would not per se be an error. But we are going to work with the fact that we will not get a space from bgen and use it as a test case.